client/src/nv_ingest_client/cli/util/processing.py

```diff
@@ -417,22 +395,56 @@ def save_response_data(response, output_directory):
     doc_map = organize_documents_by_type(response_data)
     for doc_type, documents in doc_map.items():
         doc_type_path = os.path.join(output_directory, doc_type)
-        if not os.path.exists(doc_type_path):
+        if (not os.path.exists(doc_type_path)):
             os.makedirs(doc_type_path)
 
+        if (doc_type in ("image", "structured") and images_to_disk):
+            for i, doc in enumerate(documents):
+                meta = doc.get("metadata", {})
+                image_content = meta.get("content")
+                if (doc_type == "image"):
+                    image_type = meta.get("image_metadata", {}).get("image_type", "png").lower()
+                else:
+                    image_type = "png"
+
+                if (image_content and image_type in {"png", "svg", "jpeg", "jpg", "tiff"}):
+                    try:
+                        # Decode the base64 content
+                        image_data = base64.b64decode(image_content)
+                        image = Image.open(io.BytesIO(image_data))
+
+                        # Define the output file path
+                        image_ext = "jpg" if image_type == "jpeg" else image_type
+                        image_filename = f"{clean_doc_name}_{i}.{image_ext}"
+                        image_output_path = os.path.join(doc_type_path, "media", image_filename)
+
+                        # Ensure the media directory exists
+                        os.makedirs(os.path.dirname(image_output_path), exist_ok=True)
+
+                        # Save the image to disk
+                        image.save(image_output_path, format=image_ext.upper())
+
+                        # Update the metadata content with the image path
+                        meta["content"] = os.path.realpath(image_output_path)
```

Comment:
This is the only thing I question. I do like we make it very clear the image has been decoded and saved elsewhere but then wonder if we want to just leave the base64 encoded bytes here and add another metadata field for the locaiton the file was saved?

Honestly, I can't decide on which way so I'm fine with this.

Reply:
One of the comments here is that the base64 encoded bytes make the file bad for human readability because it exceeds normal buffer sizes and doesn't scroll nicely.